name: pylsl
concurrency:
  group: ${{ github.workflow }}-${{ github.event.number }}-${{ github.event.ref }}
  cancel-in-progress: true
on:
  workflow_dispatch:
  schedule:
  - cron: '0 8 * * 1'

jobs:
  sha256:
    timeout-minutes: 2
    runs-on: ubuntu-latest
    env:
      SHA256_PYLSL: dd9222ee3894c43535b40aee542a2c8cb6062a467d4b103bd3c3a713d7c7690c
    steps:
      - name: Checkout pylsl repository
        uses: actions/checkout@v4
        with:
          repository: labstreaminglayer/pylsl
      - name: Compare checksum
        run: echo "$SHA256_PYLSL pylsl/pylsl.py" | sha256sum -c
      - name: Create GitHub issue
        if: failure()
        uses: dacbd/create-issue-action@main
        with:
          token: ${{ github.token }}
          title: pyLSL is out of sync with BSL
          body: |
            Please visit https://github.com/labstreaminglayer/pylsl and review the changes.
            [Failed Run](https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }})
          assignees: mscheltienne
